Creator Dashboard and Analytics

mdmohsin7 opened the pull request:
- [x] Capture Creator Details
- [x] Capture Payment Details
- [x] Remove name and email stuff from submit/update app
- [x] Creators dashboard screen (atleast a basic one)
- [x] Payments history
- [x] Overall apps analytics
- [x] Add manual payments
- [x] Update apps endpoints to use pydantic model



https://github.com/user-attachments/assets/6ba0bf8b-a023-4b1f-88e7-011b9f56b804

beastoin commented:
@mdmohsin7 man, what is this and why we need it ?

beastoin commented:
<img width="614" alt="Screenshot 2024-12-09 at 12 03 32" src="https://github.com/user-attachments/assets/201ed0c2-8f68-4516-96c2-768a94c71030">
and pls fix this 🥹

mdmohsin7 commented:
It doesn't work only for you @beastoin XD
Sending on iMessage, pls check

<img width="822" alt="Screenshot 2024-12-09 at 10 55 47 AM" src="https://github.com/user-attachments/assets/3f8fdf44-0aee-4989-9f6e-4ed99d68f707">

beastoin commented:
oops sry, i'm a Firefoxman. it works on safari so nvm 👀 @mdmohsin7

mdmohsin7 commented:
> @mdmohsin7 man, what is this and why we need it ?

This is for app creators to submit their paypal details, check payouts, and some basic analytics

mdmohsin7 commented:
Up for review @beastoin

beastoin commented:
**my thoughts:**

why do we need a separated creator profile man? isn't the user already a creator - or a developer - as default ? just simply put some extra fields to Developer Settings, should we?

the current analytic-related features are not stable enough(and, too small - i mean the metric) to put it into a hard-to-updates place likes app, why don't we just start with a simple web portal ?

@mdmohsin7 hope that help 🌚

mdmohsin7 commented:
> why do we need a separated creator profile man? isn't the user already a creator - or a developer - as default ? just simply put some extra fields to Developer Settings, should we?

As a simple separation between a normal user and someone who builds stuff for Omi. This is mainly to have their details like the public name, email and payment details and nothing much. The developer mode or settings has a different meaning associated with it, so to not confuse people and keep things simple I added it as a separate option. I'm fine with putting them in developer settings

> the current analytic-related features are not stable enough(and, too small - i mean the metric) to put it into a hard-to-updates place likes app, why don't we just start with a simple web portal ?

The initial version is mainly to just collect details and payout history (so that they know when we sent out the money and stuff), I added 1-2 minor analytics stuff to make it look better. Web portal would make sense when we have the create/update apps functionality on the web so that the creators use it completely instead of having to switch between platforms. Think of this inside of the app similar to the Google Play Console App on Play Store (minimal and for on the go stuff) and the web app should be similar to play console on web with all the stuff. Can get rid of the analytics but the screen would look empty as we will only have payout history and creator details

Lmk what do you think of it @beastoin

beastoin commented:
the product features related to payment and analytics have changed a lot.

/ closed
